modules/mono/glue/GodotSharp/GodotSharp/Core/GodotSynchronizationContext.cs

```diff
@@ -1,17 +1,44 @@
 using System;
 using System.Collections.Concurrent;
-using System.Collections.Generic;
 using System.Threading;
+using System.Threading.Tasks;
 
 namespace Godot
 {
     public sealed class GodotSynchronizationContext : SynchronizationContext, IDisposable
     {
-        private readonly BlockingCollection<KeyValuePair<SendOrPostCallback, object>> _queue = new();
+        private readonly BlockingCollection<(SendOrPostCallback d, object state)> _queue = new();
```

Comment:
`ValueTuple` is just a the more modern way to create a pair than `KeyValuePair` which was used before .NET had implemented tuples. They are functionally the same (in the case of a pair) but offer [nicer syntax and useful features](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/value-tuples).

The biggest benefits here are reducing the noise in the type signature and allowing for naming of the individual fields, but things like deconstruction and more sensible equality semantics are also nice to have up-front.

If there's some other reason you guys would prefer to use `KeyValuePair` though, definitely let me know and I can switch it back.

Reply:
> The biggest benefits here are reducing the noise in the type signature and allowing for naming of the individual fields, but things like deconstruction and more sensible equality semantics are also nice to have up-front.

It doesn't seem like we need the equality, and `KeyValuePair` also supports deconstruction. So it seems like the only benefit we are getting from using `ValueType` is the named fields.

> If there's some other reason you guys would prefer to use `KeyValuePair` though, definitely let me know and I can switch it back.

I'm OK with using `ValueType`, it's likely that we were using `KeyValuePair` since this was implemented when we were targeting .NET Framework 4.5.

If we use `ValueTuple`, make sure to follow the .NET naming conventions. The fields of the `ValueTuple` should be named using PascalCase. I think a good name would be `(SendOrPostCallback Callback, object State)`, because `d` is not very descriptive.